server/rootless.go

```diff
@@ -17,7 +21,41 @@ func hasNetworkNamespace(config *rspec.Spec) bool {
 }
 
 func makeOCIConfigurationRootless(g *generate.Generator) {
-	g.Config.Linux.Resources = nil
+	// Resource limitations requires cgroup v2 delegation (https://rootlesscontaine.rs/getting-started/common/cgroup2/).
+	if g.Config.Linux.Resources != nil {
+		// cannot control device eBPF with rootless
+		g.Config.Linux.Resources.Devices = nil
+		v2Controllers := getAvailableV2Controllers()
```

Comment:
I'd do something like
```go
if r.Memory != nil || r.CPU != nil || r.Pids != nil || r.BlockIO != nil || r.Rdma != nil || r.HugepageLimits != nil {
   v2Controllers := getAvailableV2Controllers()
   .....
}
```

so we don't call getAvailableV2Controllers() unnecessarily.

Reply:
updated